website/docs/components/pagination/partials/code/how-to-use.md

```diff
@@ -1,22 +1,200 @@
+
+Below is an example of some of these extra arguments:
+
+```handlebars
+<Hds::Pagination::Numbered
+  @totalItems={{40}}
+  @itemsPerPage={{20}}
+  @showTotalItems={{false}}
+  @showSizeSelector={{false}}
+  @pageSizes={{array 5 20 60}}
+/>
+```
+
+### Truncation
+
+When there is a large number of items and consequently the number of pages is also large, by default the component automatically "truncates" the number of visible pages (using "ellipsises"):
+
+```handlebars
+<Hds::Pagination::Numbered
+  @totalItems={{120}}
+/>
+```
+
+If necessary, it's possible to disable this functionality using a specific `@isTruncated` argument:
+
+```handlebars
+<Hds::Pagination::Numbered
+  @totalItems={{120}}
+  @isTruncated={{false}}
+/>
+```
+
+### Events handling
+
+The component exposes two callback functions that can be used to respond to specific events:
+
+```handlebars
+<Hds::Pagination::Numbered
+  @totalItems={{40}}
+  @onPageChange={{this.handlePageChange}}
+  @onPageSizeChange={{this.handlePageSizeChange}}
+/>
+```
+
+The first `onPageChange` function is invoked when a user interacts with a navigation control ("prev/next" or "page number") and so can be used to respond to a "page" change (eg. updating the list of items in the page and/or updating the routing/URL).
+
+The second `onPageSizeChange` function is invoked when a user interacts with the "size selector" and so can be used to respond to a "page size" change (eg. updating the number of items listed in the page, updating the routing/URL, and/or updating other elements in the page).
+
+### Routing/URL updates
 
-{description}
+If you want the pagination to change the URL of the page directly (eg. updating the query parameters) you need to pass the routing parameters to the component:
 
 ```handlebars
-<Hds::Alert @type="page" as |A|>
-  <A.Title>Title here</A.Title>
-  <A.Description>Description here</A.Description>
-</Hds::Alert>
+<Hds::Pagination::Numbered
+  @totalItems={{this.demoTotalItems}}
+  @itemsPerPage={{this.demoCurrentPageSize}}
+  @currentPage={{this.demoCurrentPage}}
+  @pageSizes={{this.demoPageSizes}}
+  @route={{this.demoRouteName}}
+  @queryFunction={{this.demoQueryFunctionNumbered}}
+/>
 ```
+
+where the `@queryFunction` function will be something like this:
+
+```javascript
+get demoQueryFunctionNumbered() {
+  return (page, pageSize) => {
+    return {
+      demoCurrentPage: page,
+      demoCurrentPageSize: pageSize,
+      demoExtraParam: 'hello',
+    };
+  };
+}
+```
+
+When the routing parameters are provided, the "navigation controls" are rendered as links and if the user clicks on one of them the page URL is automatically updated. This means that the component's state is persisted **outside** of the component and so its whole state **must** be "controlled" by the consumer's code (otherwise there would be confliting states).
```

Comment:
```suggestion
When the routing parameters are provided, the "navigation controls" are rendered as links and if the user clicks on one of them the page URL is automatically updated. This means that the component’s state is persisted **outside** of the component and so its whole state **must** be "controlled" by the consumer’s code (otherwise there would be conflicting states).
```